### PR TITLE
Improve `X509CA` disk implementation

### PR DESCRIPTION
**Pull request by maxlambrecht**
**Pull request check list**

- [X] Proper tests/regressions included?
- [X] Documentation updated?

**Description of change**

This pull request enhances the X509CA "disk" implementation by introducing a new configuration option called `bundle_file_path` and enabling the use of intermediate CA certificates. The `bundle_file_path` parameter is now required when the specified `cert_file_path` does not contain a self-signed (root) CA. It allows Galadriel to issue certificates based on an external chain of trust. The bundle file should contain one or more intermediate CA certificates and/or root CA certificates, forming the trust chain up to the root CA. This update ensures more flexibility in managing CA certificates and provides better support for complex trust relationships.

This PR also updates the Galadriel server to utilize the entire certificate chain issued by the X509CA as the TLS certificate for the TLS connections with the Harvesters.

In summary, the changes include:

X509CA "disk" Implementation:
- Added a new configuration option `bundle_file_path` to support external chains of trust.
- Updated the X509CA documentation to reflect the new configuration option.
- Modified the X509CA test cases to cover the new functionality.

Galadriel Server:
- Updated the server code to utilize the entire certificate chain issued by the X509CA as the TLS certificate for the TLS connections with the Harvesters.

These changes provide a more robust and flexible X509CA "disk" implementation, enabling Galadriel to handle a broader range of trust scenarios. 

**Which issue this pull requests fixes**

### CODE REVIEW

File: pkg/server/catalog/catalog_test.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ providers {
 	X509CA "disk" {
 		key_file_path = "%s"
 		cert_file_path = "%s"
+        bundle_file_path = "%s"
```
Comment (Victorblsilveira) [inline pkg/server/catalog/catalog_test.go:25]: Fix indentation?

**Review by Victorblsilveira — COMMENTED** [review]
Just some minor comments.

**Comment by maxlambrecht** [comment]
Thanks, I'll address your comments. There's something else I've just realized that I'm fixing: when there are certs in the `ca.trustBundle`, all certs are appended to the cert chain, even if they are root CAs, which is not the common practice.

**Comment by maxlambrecht** [comment]
**Update:** The last commits modify the trust bundle's handling to enhance the certificate chain creation process. Initially, the trust bundle is used to validate that the CA certificate can chain back to a trusted CA certificate. Following this verification, root CAs are filtered out from the trust bundle. The remaining trust bundle, set in `ca.trustBundle`, solely contains any necessary intermediate CAs. These intermediate CAs are then appended to the certificate chain during the issuance of new certificates. This approach ensures a streamlined and efficient creation of the certificate chain.

File: pkg/common/x509ca/disk/disk.go [hunk 2]
```diff
@@ -14,45 +14,67 @@ import (
...
+	ca.clock = config.Clock
+
 	if config.CertFilePath == "" {
-		return errors.New("certificate file path is required")
+		return errors.New(ErrCertPathRequired)
 	}
 
 	if config.KeyFilePath == "" {
-		return errors.New("private key file path is required")
+		return errors.New(ErrPrivateKeyPathReq)
```
[the remark this answers is not in the document]
Comment (maxlambrecht) [reply pkg/common/x509ca/disk/disk.go]: I also prefer the constant names with the suffix `Required`.

**Review by Victorblsilveira — APPROVED** [review]
